### PR TITLE
#10 Provided Tests for Brick API

### PR DESCRIPTION
**Pull request by serjiokov**
Provided unit test and test implementation for Brick API

### CODE REVIEW

File: bundles/org.eclipse.chronograph.base/OSGI-INF/OSGI-INF/l10n/bundle.properties [hunk 1]
```diff
@@ -0,0 +1,27 @@
+###############################################################################
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/OSGI-INF/OSGI-INF/l10n/bundle.properties:1]: This file should be removed

File: bundles/org.eclipse.chronograph.base/OSGI-INF/l10n/bundle.properties [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+# Sergei Kovalchuk <sergei.kovalchuk@arsysop.ru> - initial API and implementation
+###############################################################################
+
+Bundle-Name = Eclipse Chronograph Core
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/OSGI-INF/l10n/bundle.properties:14]: Base

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/AreaImpl.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.eclipse.chronograph.internal.api.Area;
+
+/**
+ * 
+ * Implementation of {@link Area} interface intended to store graphical
+ * object location
+ */
+public class AreaImpl implements Area {
+
+	private int x = 0;
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/AreaImpl.java:32]: final for all

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/AreaImpl.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	private int y = 0;
+	private int width;
+	private int height;
+
+	public AreaImpl(int x, int y, int width, int height) {
+		super();
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/AreaImpl.java:38]: not needed

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/BrickImpl.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+package org.eclipse.chronograph.internal.base;
+
+import org.eclipse.chronograph.internal.api.Brick;
+import org.eclipse.chronograph.internal.api.BrickContainer;
+import org.eclipse.chronograph.internal.api.Position;
+
+public class BrickImpl implements Brick {
+
+	//public static final int DRAWING_ITEM_HEIGHT = 20;
+	private String id;
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/BrickImpl.java:23]: final for all

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/GroupImpl.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import org.eclipse.chronograph.internal.api.GroupContainer;
+
+/**
+ * 
+ * Implementation of {@link Group} interface
+ *
+ */
+public class GroupImpl implements Group {
+
+	private String id;
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/GroupImpl.java:30]: final for all

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/PositionImpl.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+import org.eclipse.chronograph.internal.api.Position;
+
+/**
+ * 
+ * Implementation of {@link Position} interface intended to store period of time
+ *
+ */
+public class PositionImpl implements Position {
+	private long start;
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/PositionImpl.java:24]: final for all

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/SectionImpl.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+import org.eclipse.chronograph.internal.api.Section;
+
+/**
+ * 
+ * Implementation of {@link Section}
+ *
+ */
+public class SectionImpl implements Section {
+
+	private String id;
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/SectionImpl.java:30]: final for all

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/BrickContentProviderImpl.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+ * @param <I> - input object type
+ * @param <D> - intermediate adapted object type
+ */
+public class BrickContentProviderImpl<I, D> implements BrickContentProvider<I> {
+
+	private final BrickBuilder<List<I>, D> builder;
+	private final BrickAdapter<D, Brick> adapter;
+
+	public BrickContentProviderImpl(BrickBuilder<List<I>, D> brickBuilder, BrickAdapter<D, Brick> brickAdapter) {
+
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/BrickContentProviderImpl.java:37]: redundant line

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/ChronographProviderImpl.java [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+ * @param <I> - provider type
+ */
+public class ChronographProviderImpl<I> implements ConteinerProvider<I> {
+
+	private final SectionContentProvider<I> sectionProvider;
+	private final GroupContentProvider<I> groupProvider;
+	private final SubGroupContentProvider<I> subGroupProvider;
+	private final BrickContentProvider<I> brickProvider;
+	private final StageLabelProvider labelProvider;
+	private List<I> input;
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/ChronographProviderImpl.java:38]: final

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/ChronographProviderImpl.java [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+	public ChronographProviderImpl(List<I> input, SectionContentProvider<I> sectionProvider, //
+			GroupContentProvider<I> groupProvider, //
+			SubGroupContentProvider<I> subGroupProvider, //
+			BrickContentProvider<I> brickProvider, //
+			StageLabelProvider labelProvider) {
+		super();
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/ChronographProviderImpl.java:45]: not needed

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/ChronographProviderImpl.java [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+		return labelProvider;
+	}
+
+	@Override
+	public List<I> getInput() {
+		return input;
+	}
+
+	@Override
+	public void updateInput(List<I> input) {
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/ChronographProviderImpl.java:85]: Please create a ticket to rework this

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/ChronographProviderImpl.java [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+import org.eclipse.chronograph.internal.api.providers.StageLabelProvider;
+import org.eclipse.chronograph.internal.api.providers.SubGroupContentProvider;
+
+/**
+ * 
+ * Aggregated provider implementation of {@link ConteinerProvider}
+ *
+ * @param <I> - provider type
+ */
+public class ChronographProviderImpl<I> implements ConteinerProvider<I> {
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/ChronographProviderImpl.java:31]: `ConteinerProvider` should be renamed in another PR

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/SectionContentProviderImpl.java [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+ * @param <I> - input object type
+ * @param <D> - intermediate adapted object type
+ */
+public class SectionContentProviderImpl<I, D> implements SectionContentProvider<I> {
+
+	private final SectionBuilder<List<I>, D> builder;
+	private final SectionAdapter<D, Section> adapter;
+
+	public SectionContentProviderImpl(SectionBuilder<List<I>, D> sectionBuilder, SectionAdapter<D, Section> adapter) {
+
```
Comment (ruspl-afed) [inline bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/providers/SectionContentProviderImpl.java:36]: unnecessary line

File: tests/org.eclipse.chronograph.api.tests/META-INF/MANIFEST.MF [hunk 15]
```diff
@@ -0,0 +1,12 @@
+Manifest-Version: 1.0
+Automatic-Module-Name: org.eclipse.chronograph.api.test
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/META-INF/MANIFEST.MF:2]: org.eclipse.chronograph.base.tests

File: tests/org.eclipse.chronograph.api.tests/META-INF/MANIFEST.MF [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+Bundle-ManifestVersion: 2
+Bundle-SymbolicName: org.eclipse.chronograph.api.tests
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/META-INF/MANIFEST.MF:4]: org.eclipse.chronograph.base.tests

File: tests/org.eclipse.chronograph.api.tests/OSGI-INF/l10n/bundle.properties [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+# terms of the Eclipse Public License 2.0 which is available at
+# https://www.eclipse.org/legal/epl-2.0/.
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#    Sergei Kovalchuk <sergei.kovalchuk@arsysop.ru> - initial API and implementation
+###############################################################################
+
+Bundle-Name = Chronograph API Base Tests
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/OSGI-INF/l10n/bundle.properties:15]: Eclipse Chronograph Base Tests

File: tests/org.eclipse.chronograph.api.tests/build.properties [hunk 18]
```diff
@@ -0,0 +1,5 @@
+source.. = src/
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/build.properties:1]: Header missed

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/brick/test/BrickProvidersTest.java [hunk 19]
```diff
@@ -0,0 +1,82 @@
+package org.eclipse.chronograph.api.brick.test;
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/brick/test/BrickProvidersTest.java:1]: Header missed

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/brick/test/TestBrickAdapter.java [hunk 20]
```diff
@@ -0,0 +1,32 @@
+package org.eclipse.chronograph.api.brick.test;
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/brick/test/TestBrickAdapter.java:1]: Header missed

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/brick/test/TestBrickBuilder.java [hunk 21]
```diff
@@ -0,0 +1,29 @@
+package org.eclipse.chronograph.api.brick.test;
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/brick/test/TestBrickBuilder.java:1]: Header missed

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/group/test/GroupProvidersTest.java [hunk 22]
```diff
@@ -0,0 +1,42 @@
+package org.eclipse.chronograph.api.group.test;
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/group/test/GroupProvidersTest.java:1]: Header missed

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/group/test/TestGroupAdapter.java [hunk 23]
```diff
@@ -0,0 +1,30 @@
+package org.eclipse.chronograph.api.group.test;
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/group/test/TestGroupAdapter.java:1]: Header missed

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/group/test/TestGroupBuilder.java [hunk 24]
```diff
@@ -0,0 +1,21 @@
+package org.eclipse.chronograph.api.group.test;
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/group/test/TestGroupBuilder.java:1]: Header missed

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/test/TestInputObject.java [hunk 25]
```diff
@@ -0,0 +1,56 @@
+package org.eclipse.chronograph.api.test;
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/test/TestInputObject.java:1]: Header missed

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/test/TestInputObjects.java [hunk 26]
```diff
@@ -0,0 +1,26 @@
+package org.eclipse.chronograph.api.test;
```
Comment (ruspl-afed) [inline tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/test/TestInputObjects.java:1]: Header missed

File: bundles/org.eclipse.chronograph.base/OSGI-INF/OSGI-INF/l10n/bundle.properties (hunk 1 above)
Comment (serjiokov) [reply bundles/org.eclipse.chronograph.base/OSGI-INF/OSGI-INF/l10n/bundle.properties]: File removed.

File: bundles/org.eclipse.chronograph.base/OSGI-INF/l10n/bundle.properties (hunk 2 above)
Comment (serjiokov) [reply bundles/org.eclipse.chronograph.base/OSGI-INF/l10n/bundle.properties]: Renamed.

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/AreaImpl.java (hunk 3 above)
Comment (serjiokov) [reply bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/AreaImpl.java]: Applied "final" field modificator.

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/BrickImpl.java (hunk 5 above)
Comment (serjiokov) [reply bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/BrickImpl.java]: Applied "final" field modificator.

File: bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/AreaImpl.java (hunk 4 above)
Comment (serjiokov) [reply bundles/org.eclipse.chronograph.base/src/org/eclipse/chronograph/internal/base/AreaImpl.java]: Removed.

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/group/test/TestGroupAdapter.java (hunk 23 above)
Comment (serjiokov) [reply tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/group/test/TestGroupAdapter.java:22]: The header added.

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/group/test/TestGroupBuilder.java (hunk 24 above)
Comment (serjiokov) [reply tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/group/test/TestGroupBuilder.java:22]: The header was added.

File: tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/test/TestInputObjects.java (hunk 26 above)
Comment (serjiokov) [reply tests/org.eclipse.chronograph.api.tests/src/org/eclipse/chronograph/api/test/TestInputObjects.java:22]: The header was added.
